drivers/soundwire/intel.c

```
@@ -1554,7 +1554,7 @@ static int __maybe_unused intel_pm_prepare(struct device *dev)
 	struct sdw_intel *sdw = cdns_to_intel(cdns);
 	struct sdw_bus *bus = &cdns->bus;
 	u32 clock_stop_quirks;
-	int ret = 0;
+	int ret;
```

Hrm, not the most robust error handling if I look further in `intel_pm_prepare()`
All those dev_err() prints should be dev_warn() as the fails are not treated as errors?

---

they are errors all right but there's a comment saying why it's ignored so might just as well keep the code as is.

---

Fair enough, but since we ignore the error (not treating it like an error) it might as well should not be an error print as it will magically going to be fixed on resume.

While reading the comment explaining: this:
```c
/*
 * If any operation in this block fails, we keep going since we don't want
 * to prevent system suspend from happening and errors should be recoverable
 * on resume.
 */
```
The first error handling is returning 0 in spirit of `If any operation in this block fails, we keep going` ;)

The last error print is not needed as the `intel_resume_child_device` will print a more detailed print?

Anyways this is unrelated to this PR.

---

I agree in general @ujfalusi, the main reason why I kept the errors is also that we have these random failures on some SoundWire devices and I really want the CI to flag those cases so that we finally figure out what happens. I have several other patches that flag errors but that we kept in the SOF tree.